### PR TITLE
Use setCache to get rid of messy loadMoreChildrenImpl logic

### PR DESCRIPTION
**Pull request by nturinski**
Relies on https://github.com/microsoft/vscode-azuretools/pull/1142

I was using a bunch of weird flags/state to effectively use our cache.  I really didn't like that implementation because it seems like it could be very error prone.  If you try to refresh and change grouping or focus, it was having some weird behaviors.

`loadMoreChildren` tends to get called a lot too so there was a good chance that two separate calls could interfere with each other.  Changing to the setCache method should be safer since all calls are synchronous and it is never automatically called by the VS Code TreeProvider API.

I considered using the same approach for `GroupTreeItemBase.ts` but it seemed a bit overkill to register an event for it since the loadMoreChildrenImpl is already synchronous anyway.

### CODE REVIEW

File: src/tree/SubscriptionTreeItem.ts [hunk 1]
```diff
@@ -246,6 +246,14 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
         return this.cache.treeMaps[groupBy] ?? {};
     }
 
+    private _setCachedChildren(childrenToSet: AzExtTreeItem[]): void {
+        // To access the private cacheChildren, go buckwild and ignore typings!!
+        const thisUnknown = this as unknown as { _cachedChildren: AzExtTreeItem[] };
```
Comment (bwateratmsft) [inline src/tree/SubscriptionTreeItem.ts:251]: image

It's not my fault, I have young children.

Comment (nturinski) [reply src/tree/SubscriptionTreeItem.ts:251]: I'm pretty sure that this is Frozen 2... Deep cut.

Comment (alexweininger) [reply src/tree/SubscriptionTreeItem.ts:251]: 🤣

**Comment by alexweininger** [comment]
Have you verified this fixes #208 and #213 ?

File: package.json [hunk 2]
```diff
@@ -476,7 +476,7 @@
         "@azure/arm-resources": "5.0.0",
         "@azure/arm-resources-profile-2020-09-01-hybrid": "^1.0.0",
         "@microsoft/vscode-azext-azureutils": "^0.2.0",
-        "@microsoft/vscode-azext-utils": "^0.2.3",
+        "@microsoft/vscode-azext-utils": "0.2.4",
```
Comment (bwateratmsft) [inline package.json:479]:
```suggestion
        "@microsoft/vscode-azext-utils": "^0.2.4",
```

File: src/tree/SubscriptionTreeItem.ts [hunk 3]
```diff
@@ -92,48 +92,29 @@ export class SubscriptionTreeItem extends SubscriptionTreeItemBase {
...
+            // await Promise.all(applicationResourceProviders.map((provider: ApplicationResourceProvider) => async () => this.rgsItem.push(...(await provider.provideResources(this.subscription) ?? []))));
+
+            resources.forEach(item => ext.activationManager.onNodeTypeFetched(item.type));
+            this.cache.appResources = resources.map((resource: AppResource) => AppResourceTreeItem.Create(this, resource));
+        }
+        await this.createTreeMaps(clearCache, context);
+
+        const focusedGroupId = await ext.context.workspaceState.get('focusedGroup') as string;
+        // on first load, check if there was persistent setting
+        const focusGroupTreeItem = await this.tryGetFocusGroupTreeItem(focusedGroupId);
```
Comment (bwateratmsft) [inline src/tree/SubscriptionTreeItem.ts:112]: Can we have `tryGetFocusGroupTreeItem` internally take care of looking up the setting value?

**Comment by nturinski** [comment]
Closing in favor of https://github.com/microsoft/vscode-azureresourcegroups/pull/222

**Pull request closed without merging**